executor/ddl.go

```diff
@@ -635,6 +640,44 @@ func (e *DDLExec) executeRecoverTable(s *ast.RecoverTableStmt) error {
 	return err
 }
 
+// recoverTablePlacement is used when recover/flashback table.
+// It will replace the placement policy of table with the direct options because the original policy may be deleted
+func recoverTablePlacement(ctx sessionctx.Context, snapshotTS uint64, tblInfo *model.TableInfo) (*model.TableInfo, error) {
+	is, err := domain.GetDomain(ctx).GetSnapshotInfoSchema(snapshotTS)
+	if err != nil {
+		return nil, err
+	}
+
+	if ref := tblInfo.PlacementPolicyRef; ref != nil {
+		policy, ok := is.PolicyByName(ref.Name)
+		if !ok || policy.ID != ref.ID {
```

Comment:
I think we need to take care of the skip placmeent validation option here? And recover the table anyway, if we really can not recover the placement rules?

Reply:
Actually, only bug can cause this failure. But, yes... , skip the validation may be better in practice to avoid some unexpected situations. What do you think about it? @morgo

Reply:
Can not you trigger it if you drop the policy?

Reply:
What do you mean by "trigger it if you drop the policy" ?

Reply:
This is is the snapshot one. So It should have the policy if there is no bug.